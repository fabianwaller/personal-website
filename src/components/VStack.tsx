import { cn } from "@/lib/utils";

type VStackProps = {
  children: React.ReactNode;
  className?: string;
  narrow?: boolean;
};

const VStack: React.FC<VStackProps> = ({ children, className, narrow }) => (
  <div
    className={cn([
      "flex flex-col items-center justify-center gap-4",
      narrow && "gap-2",
      className,
    ])}
  >
    {children}
  </div>
);

export default VStack;
