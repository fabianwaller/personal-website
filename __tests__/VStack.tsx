import VStack from "@/components/VStack";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

describe("VStack", () => {
  it("renders VStack unchanged", () => {
    const component = render(
      <VStack>
        <p>1</p>
        <p>2</p>
      </VStack>,
    );
    expect(component.asFragment()).toMatchSnapshot();
  });
});
