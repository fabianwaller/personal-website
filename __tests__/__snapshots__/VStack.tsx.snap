// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VStack renders VStack unchanged 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center gap-4"
  >
    <p>
      1
    </p>
    <p>
      2
    </p>
  </div>
</DocumentFragment>
`;
